scripts/shellcheck.sh

```diff
@@ -0,0 +1,253 @@
+
+Usage: ./scripts/shellcheck.sh [-e|--exclude-code=CODE] [-S|--severity=LEVEL] [-v|--view]
+
+example: ./scripts/shellcheck.sh -e SC1091 -e SC1090 -S warning -v
+
+EOF
+}
+
+parse_cli_args() {
+    echo; echo "Command line arguments: $0 $*"; echo
+
+    SEVERITY=
+    EXCLUDE_CODES=
+    VIEW=
+
+    _help=
+    _invalid_arg_count=0
+    while [[ $# -gt 0 ]]; do
+        optarg=
+        _err=0
+        shft_cnt=1
+        if [[ "$1" = '--' ]]; then
+            break
+        elif [[ "$1" =~ ^-- && $# -gt 1 && ! "$2" =~ ^- ]]; then
+            optarg="$2"
+            shft_cnt=2
+        elif [[ "$1" =~ ^-[^-] && $# -gt 1 && ! "$2" =~ ^- ]]; then
+            optarg="$2"
+            shft_cnt=2
+        elif [[ "$1" =~ ^-[^-] ]]; then
+            optarg="${1/??/}"
+        fi
+
+        case "$1" in
+            -S*|--severity*)
+                # don't let the user enter -S LEVEL more than once
+                if [[ -n "$SEVERITY" ]]; then
+                    printf "${RED}[Error]${RST_COLOR} Severity level already set to: '${CYAN}%s${RST_COLOR}'\n" "${SEVERITY}"
+                    _err=1
+                # valid if -S has any of 'info', 'warning', 'error'
+                elif [[ "${optarg}" == "info" || "${optarg}" == "warning" || "${optarg}" == "error" ]]; then
+                    SEVERITY="${optarg}"
+                else
+                    # continue and disregard invalid severity level
+                    printf "${RED}[Error]${RST_COLOR} invalid severity level: '${CYAN}%s${RST_COLOR}'\n" "${optarg}"
+                    _err=1
+                fi
+                shift "$shft_cnt"
+                ;;
+            -e*|--exclude-code*)
+                # strip whitespace from optarg
+                CODE=$(echo "${optarg}" | tr -d '[:space:]')
+                # valid if matching format SC1000-SC9999
+                if [[ "${optarg}" =~ ^SC[0-9]{4}$ ]]; then
+                    # if empty then populate with just error code, otherwise add pipe before new code for grep later
+                    if [[ -z "${EXCLUDE_CODES}" ]]; then
+                        EXCLUDE_CODES+="${CODE}"
+                    else
+                        EXCLUDE_CODES+=",${CODE}"
+                    fi
+                else
+                    # continue just don't save invalid error code
+                    printf "${RED}[Error]${RST_COLOR} Invalid error code entered: %s\n" "${optarg}"
+                    _err=1
+                fi
+                shift "$shft_cnt"
+                ;;
+            -v|--view)
+                VIEW="True"
+                shift "$shft_cnt"
+                ;;
+            -h|--help)
+                _help=1
+                ;;
+            *)
+                printf "${RED}[Error]${RST_COLOR} Unrecognized option: %s\n" "$1"
+                _err=1
+                shift "$shft_cnt"
+                ;;
+        esac
+
+        _invalid_arg_count=$(( "${_invalid_arg_count}" + 1 ))
```

Comment:
```suggestion
        _invalid_arg_count=$(( _invalid_arg_count + 1 ))
```
In bash, there is a shortcut for accessing variable contents without `$` in arithmetic contexts; and, in fact, this tends to behave better. When you don't need to rely on parameter expansion, this is a good plan.

Reply:
It should be actually at the before stage to not always += 1.         
```console
_invalid_arg_count=$(( "${_invalid_arg_count}" + "${_err}" ))
```
ACK, changing to
```console
_invalid_arg_count=$(( _invalid_arg_count + _err ))
```